### DISCUSSION TITLE
Regarding the limitation of ezkl.

### DISCUSSION THREAD

**Discussion by saeyoon17** [zkonduit/ezkl#488] [Q&A] [answered]
Hi, it was so nice to see such a nice library. I really liked this line of work and wanted to ask a question.
1. What are some current bottlenecks of ezkl? (Limited Ops? UI?)
2. What are the future directions?

Again, I really appreciate the work :)

**Comment by saeyoon17** [comment]
+) Would it help if I organize the examples in terms of ML model types and make pull request? (NLP, Vision)
I believe doing so will also benefit in the future with much more examples.

**Reply by jasonmorton** [reply]
Sure, that would be wonderful. Would you like to start by adding a new notebook example with a model you are interested in?

**Reply by saeyoon17** [reply]
@jasonmorton Great! If it's okay, I have a few more questions regarding the examples. 
- Some do have training phase (using pytorch-lightning) and some do not. Does training phase needs to be included?
- Is it okay to not use pytorch-lightning in the training phase for example notebook?

Since there are no GNNs yet, I think I might start with building GNN examples. :)

**Reply by jasonmorton** [reply, marked as answer]
Whether there is a training phase is up to you. It makes the example more realistic and forkable, but also longer. Use whatever you like for training, as long as it will run in Google colab. GNN, sounds great!
